exec/call_test.go

```diff
@@ -153,3 +190,50 @@ func TestHostSymbolCall(t *testing.T) {
 		t.Fatalf("Did not get the right value. Got %d, wanted %d", rtrns, 3)
 	}
 }
+
+func TestGoFunctionCallChecksForFirstArgument(t *testing.T) {
+	m, err := wasm.ReadModule(bytes.NewReader(moduleCallHost), invalidImporter)
+	if err != nil {
+		t.Fatalf("Could not read module: %v", err)
+	}
+	defer func() {
+		if r := recover(); r == nil {
+			t.Errorf("This code should have panicked.")
+		} else {
+			if r != "exec: the first argument of a host function was int32, expected ptr" {
```

Comment:
this looks brittle.
but I must admit we don't have a safer mechanism in place to handle this kind of thing...